shared/AppInsightsCommon/src/StorageHelperFuncs.ts

```diff
@@ -30,24 +31,55 @@ function _getLocalStorageObject(): Storage {
  * @returns {Storage} Returns storage object verified that it is usable
  */
 function _getVerifiedStorageObject(storageType: StorageType): Storage {
+    let result = null;
+    const storageTypeName = storageType === StorageType.LocalStorage ? "localStorage" : "sessionStorage";
+    
     try {
-        if (isNullOrUndefined(getGlobal())) {
-            return null;
+        // Default to false - assume storage is not available
+        let canAccessStorage = false;
+        const gbl = getGlobal();
+        
+        // Only proceed if we have a global object
+        if (!isNullOrUndefined(gbl)) {
+            // Try the safest method first (property descriptor)
+            try {
+                const descriptor = objGetOwnPropertyDescriptor(gbl, storageTypeName);
+                if (descriptor && descriptor.get) {
+                    canAccessStorage = true;
+                }
+            } catch (e) {
+                // If descriptor check fails, try direct access
+                try {
+                    canAccessStorage = !!gbl[storageTypeName];
```

Comment:
Any reason to have this in a try-catch if it's already in the scope of another try-catch when it'll just fail silently and eat the exception here?